src/openrct2/park/ParkFile.cpp

```diff
@@ -386,22 +399,107 @@ namespace OpenRCT2
 
                 ReadWriteStringTable(cs, gScenarioDetails, "en-GB");
 
-                cs.ReadWrite(gScenarioObjective.Type);
-                cs.ReadWrite(gScenarioObjective.Year);
-                cs.ReadWrite(gScenarioObjective.NumGuests);
-                cs.ReadWrite(gScenarioObjective.Currency);
-
-                cs.ReadWrite(gScenarioParkRatingWarningDays);
-
-                cs.ReadWrite(gScenarioCompletedCompanyValue);
-                if (gScenarioCompletedCompanyValue == MONEY64_UNDEFINED
-                    || gScenarioCompletedCompanyValue == COMPANY_VALUE_ON_FAILED_OBJECTIVE)
+                if (os.GetMode() == OrcaStream::Mode::READING)
```

Comment:
This doesn't make much sense. You should have one versions of the reading code for legacy version and otherwise it should be a ReadWrite.

Reply:
This is NOT what I wanted, but I was told to leave out saving to the new system. So in order to make sure it can still save to the old system, again, I had to create new code to do just that, convert it back to the old system.